trio/_core/_generated_io_windows.py

```diff
@@ -8,77 +8,87 @@
 
 from ._ki import LOCALS_KEY_KI_PROTECTION_ENABLED
 from ._run import GLOBAL_RUN_CONTEXT
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, ContextManager
+
+if TYPE_CHECKING:
+    import socket
+    from ._windows_cffi import Handle, CData
+    from typing_extensions import Buffer
+
+    from ._unbounded_queue import UnboundedQueue
 import sys
 
 assert not TYPE_CHECKING or sys.platform=="win32"
 
 
-async def wait_readable(sock):
+async def wait_readable(sock: (socket.socket | int)) ->None:
```

Comment:
In https://github.com/python-trio/trio/pull/2774/files#diff-175d00891570043e8c2954032ac59ba7aca294153052943cb309e07460969d74 I'm changing these to accept `_HasFileNo` instead of `socket.socket` - might as well do that directly here (and all other methods accepting `socket.socket`)